### ISSUE TITLE
As a user, I want to search by a street address so that I can quickly zoom into and view the area I'm interested in

### ISSUE THREAD

**Issue by katherinedm-usds** [usds/justice40-tool#384]
**Description**
Particularly in areas of the country where there might not be a lot of orienting features to help navigating to and zooming into a particular census block group, we need to provide the ability to more easily zoom into a particular address.

**Solution**
Search bar that visitor can add street address, or partial address such as street, city, zip code, similar behavior to Google Maps, such that the tool will zoom to show the vicinity of that address. For street address, it is not necessary (for this user story) to automatically select the CBG that contains the street address. 

**Describe alternatives you've considered**

**Links to user research or other resources**
 - @janelien to confirm design, in particular zoom level or area shown if only partial address (street name without street number or only zip code) is entered 
- related "help wanted" https://app.zenhub.com/workspaces/justice40-60993f6e05473d0010ec44e3/issues/usds/justice40-tool/197

**Tasks**
- [ ] @janelien confirm search bar design
- [ ] @janelien confirm design/experience for valid complete street address or partial street address (street name with city or zip code, or just zip code) is entered in search bar - for instance will we just pick a particular zoom level that enables making CBG selection easier even if it's the same zoom level for either full street address and zip code?
- [ ] @janelien confirm expected behavior if invalid address is entered
- [ ] @janelien confirm expected behavior if multiple areas are valid for the same address (edge cases where maybe there's two times the same street shows up in one city, etc.)
 - [ ] Dev implement search bar
 - [ ] Dev implement search functionality 
 - [ ] Dev implement functionality to zoom in
 - [ ] Dev implement main edge case behavior - invalid address and if multiple areas could be shown for the same address

**Definition of "Done"**
Search for valid and invalid addresses in search bar works as expected

**Comment by NatHillardUSDS** [comment]
See more notes on dev feasibility / recommendations from the spike ticket [here](https://app.zenhub.com/workspaces/justice40-60993f6e05473d0010ec44e3/issues/usds/justice40-tool/197)

**Comment by NatHillardUSDS** [comment]
More info from the spike ticket: 
Nominatim has two different kinds of searches:
a. Free-form query-based searches: the app passes a `q` parameter, which is a free-text field that can have any kind of entity in it
b. Structured search: the app passes one of 6 entities: `street`, `city`, `county`, `state`, `country`, and `postalcode`

The docs explicitly call out that we should not mix the query-based and the structured methods. 

To note: though structured is faster and more accurate, it gives less flexibility on the front-end, in that we require users to know how to categorize the info they provide. 

In the short term we decided to go with free-text string and measure / document, and to provide a flag to switch to zip-only search if needbe

**Comment by janelien** [comment]
**TL;DR - We can search by city, county, address (mostly), and zipcode and decided as a group to have the search field be freeform; hint text would be "Search by city, county, address, or zipcode"**

Beth, Katherine, and I met to try out search terms using the Mapping for EJ map as a proxy for what our search could be if it was a freeform field. Nat joined us for a bit and helped troubleshoot. Found out a couple things:
1) Search does not work on OMB laptops. There's an error that needs further research. This could potentially be problematic for program officers trying to use search.
2) Learned from Nat that some of the OpenStreetMap info is user generated (some info is coming from companies), so some areas are better "documented" than others.
3) Freeform search is not that bad.
Here is a list of our search terms with what worked/didn't, thanks to Beth's knowledge of the area:
St. Paul, Virginia - kind of works
St. Paul, VA - did not work
24283 - works
Appalachia - did not work (not surprising)
St. Paul - did not work
Wise County - works
Wise County, VA - works
St Paul - does not work
Clinch river - works (surprisingly)
[a more rural address] 3025 4th Ave St. Paul, VA - does not work
3025 4th Ave St. Paul, VA 24283 - does not work
3025 4th Ave - does not work
Western Front Hotel - did not work
3025 Fourth Avenue - does not work
4th Ave and Broad St - does not work
Carytown - works
Carytown, Richmond - works
[a more urban address] 3109 W Cary St, Richmond, VA 23221 - works
Southwest Virginia - does not work
Richmond, VA - works

**Comment by katherinedm-usds** [comment]
Moving back to the product backlog column, though it's not clear to me whether federal program officer user testing is really required in order to move forward with the solutions Nat mentions in 2 and 3. I think we can test search using example grants/benefits that have already been awarded by the Justice40 pilot programs. The BRIC example @KameronKerger can step through with the team is a great example. In addition we can see if internal USDS folks can help test for us based on either geographic areas they are familiar with, or areas that their projects are focused on.

**Comment by BethMattern** [comment]
We need to do research with agency program managers to assess the priority of search before moving this ticket forward.

**Comment by katherinedm-usds** [comment]
tracking work via issue https://app.zenhub.com/workspaces/justice40-60993f6e05473d0010ec44e3/issues/usds/justice40-tool/909
